PhysicsTools/Heppy/python/physicsobjects/Electron.py

```diff
@@ -19,579 +23,117 @@ def _physObjInit(self):
-                return self._mvaRun2[name]
-            if name not in ElectronMVAID_ByName: raise RuntimeError, "Unknown electron run2 mva id %s (known ones are: %s)\n" % (name, ElectronMVAID_ByName.keys())
-            if name in ("Spring16HZZ","Spring16GP","Fall17noIso","Fall17Iso"):
-                if self.event == None: raise RuntimeError, "You need to set electron.event before calling any new MVA"
-                self._mvaRun2[name] = ElectronMVAID_ByName[name](self.physObj, self.event, self.associatedVertex, self.rho, debug)
-            else:
-                if self.associatedVertex == None: raise RuntimeError, "You need to set electron.associatedVertex before calling any MVA"
-                if self.rho              == None: raise RuntimeError, "You need to set electron.rho before calling any MVA"
-                # -v---- below is correct in Heppy 74X, but probably not functional anyway
-                self._mvaRun2[name] = ElectronMVAID_ByName[name](self.physObj, self.associatedVertex, self.rho, True, debug)
-                # -v---- below would be correct for CMGTools 74X witht the updated Spring15 MVA electron ID
-                #if self.event            == None: raise RuntimeError, "You need to set electron.event before calling any MVA"
-                #self._mvaRun2[name] = ElectronMVAID_ByName[name](self.physObj, self.associatedVertex, self.event, self.rho, True, debug)
-        return self._mvaRun2[name]
-
-    def mvaIDTight(self, full5x5=False):
-            eta = abs(self.superCluster().eta())
-            if self.pt() < 20:
-                if   (eta < 0.8)  : return self.mvaTrigV0(full5x5) > +0.00;
-                elif (eta < 1.479): return self.mvaTrigV0(full5x5) > +0.10;
-                else              : return self.mvaTrigV0(full5x5) > +0.62;
-            else:
-                if   (eta < 0.8)  : return self.mvaTrigV0(full5x5) > +0.94;
-                elif (eta < 1.479): return self.mvaTrigV0(full5x5) > +0.85;
-                else              : return self.mvaTrigV0(full5x5) > +0.92;
-
-    def mvaIDLoose(self, full5x5=False):
-            eta = abs(self.superCluster().eta())
-            if self.pt() < 10:
-                if   (eta < 0.8)  : return self.mvaNonTrigV0(full5x5) > +0.47;
-                elif (eta < 1.479): return self.mvaNonTrigV0(full5x5) > +0.004;
-                else              : return self.mvaNonTrigV0(full5x5) > +0.295;
+        self._mvaid_category = {}
+        self._mvaid_score = {}
+        self._mvaid_normscore = {}
+        self._mvaid_passed = {}
+
+    def mva_category(self, name):
+        '''takes an mva name and returns the category of the electron,
+        only available using FWLite and Egamma code.'''
+        if 'cutBased' in name :
+            return None
+        if name not in self._mvaid_category :
+            category = self._mva_score_and_category(name)[1]
```

Comment:
shouldn't this be `self._mvaid_category[name] = self._mva_score_and_category(name)[1]` ?